test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_wait.cpp

```diff
@@ -0,0 +1,395 @@
+        err = clEnqueueWaitSemaphoresKHR(queue, 1, semaphore, nullptr, 1,
+                                         nullptr, nullptr);
+        test_failure_error(err, CL_INVALID_EVENT_WAIT_LIST,
+                           "Unexpected clEnqueueWaitSemaphoresKHR return");
+
+        // Create user event
+        clEventWrapper user_event = clCreateUserEvent(context, &err);
+        test_error(err, "Could not create user event");
+
+        // (2) Wait semaphore
+        err = clEnqueueWaitSemaphoresKHR(queue, 1, semaphore, nullptr, 0,
+                                         &user_event, nullptr);
+
+        cl_int signal_error = clSetUserEventStatus(user_event, CL_COMPLETE);
+        test_error(signal_error, "clSetUserEventStatus failed");
+
+        test_failure_error(err, CL_INVALID_EVENT_WAIT_LIST,
+                           "Unexpected clEnqueueWaitSemaphoresKHR return");
+
+        // (3) Wait semaphore
+        cl_event wait_list[] = { nullptr, nullptr, nullptr };
+        err = clEnqueueWaitSemaphoresKHR(
+            queue, 1, semaphore, nullptr,
+            sizeof(wait_list) / sizeof(wait_list[0]), wait_list, nullptr);
+        test_failure_error(err, CL_INVALID_EVENT_WAIT_LIST,
+                           "Unexpected clEnqueueWaitSemaphoresKHR return");
+
+        return CL_SUCCESS;
+    }
+};
+
+// the execution status of any of the events in event_wait_list is a negative
+// integer value.
+
+struct WaitInvalidEventStatus : public SemaphoreTestBase
+{
+    WaitInvalidEventStatus(cl_device_id device, cl_context context,
+                           cl_command_queue queue)
+        : SemaphoreTestBase(device, context, queue)
+    {}
+
+    cl_int Run() override
+    {
+        // Create semaphore
+        cl_semaphore_properties_khr sema_props[] = {
+            static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_KHR),
+            static_cast<cl_semaphore_properties_khr>(
+                CL_SEMAPHORE_TYPE_BINARY_KHR),
+            0
+        };
+
+        cl_int err = CL_SUCCESS;
+        semaphore =
+            clCreateSemaphoreWithPropertiesKHR(context, sema_props, &err);
+        test_error(err, "Could not create semaphore");
+
+        // Signal semaphore
+        err = clEnqueueSignalSemaphoresKHR(queue, 1, semaphore, nullptr, 0,
+                                           nullptr, nullptr);
+        test_error(err, "Could not signal semaphore");
+
+        // Create user event
+        clEventWrapper user_event = clCreateUserEvent(context, &err);
+        test_error(err, "Could not create user event");
+
+        // Now release the user event, which will allow our actual action to run
+        err = clSetUserEventStatus(user_event, -1);
+        test_error(err, "Unable to set event status");
+
+        // Wait semaphore
+        err = clEnqueueWaitSemaphoresKHR(queue, 1, semaphore, nullptr, 1,
+                                         &user_event, nullptr);
+        test_failure_error(err, CL_INVALID_EVENT_WAIT_LIST,
```

Comment:
As per [spec](https://registry.khronos.org/OpenCL/specs/3.0-unified/pdf/OpenCL_API.pdf), shouldn't this be **CL_EXEC_STATUS_ERROR_FOR_EVENTS_IN_WAIT_LIST**?

Reply:
Corrected, thanks!